### PR TITLE
fix: 应用授权不会自动推送的bug

### PR DESCRIPTION
**Pull request by fit2bot**
fix: 应用授权不会自动推送的bug

### CODE REVIEW

**Review by ibuler — APPROVED** [review]

File: apps/perms/signals_handler.py [hunk 1]
```diff
@@ -244,3 +245,84 @@ def on_node_asset_change(action, instance, reverse, pk_set, **kwargs):
...
+        if asset_id:
+            assets_id.append(asset_id)
+
+    for system_user in system_users:
+        system_user.assets.add(*assets_id)
+        if system_user.username_same_with_user:
+            users_id = instance.users.all().values_list('id', flat=True)
+            groups_id = instance.user_groups.all().values_list('id', flat=True)
+            system_user.groups.add(*users_id)
+            system_user.users.add(*groups_id)
```
Comment (BaiJiangJie) [inline apps/perms/signals_handler.py:273]: system_user.groups.add(*users_id)
 system_user.users.add(*groups_id)
写反了

File: apps/perms/signals_handler.py [hunk 2]
```diff
@@ -244,3 +245,84 @@ def on_node_asset_change(action, instance, reverse, pk_set, **kwargs):
...
+    groups_id = UserGroup.objects.filter(pk__in=pk_set).values_list('id', flat=True)
+    system_users = instance.system_users.all()
+
+    for system_user in system_users:
+        if system_user.username_same_with_user:
+            system_user.groups.add(*groups_id)
+
+
+@receiver(m2m_changed, sender=ApplicationPermission.applications.through)
+def on_remoteapps_permission_user_groups_changed(sender, instance, action, reverse, pk_set, **kwargs):
```
Comment (BaiJiangJie) [inline apps/perms/signals_handler.py:311]: on_remoteapps_permission_user_groups_changed
on_remoteapps_permission_user_groups_changed
和上面的名字重复了

File: apps/perms/signals_handler.py [hunk 3]
```diff
@@ -244,3 +245,84 @@ def on_node_asset_change(action, instance, reverse, pk_set, **kwargs):
         node_pk_set = pk_set
 
     create_rebuild_user_tree_task_by_related_nodes_or_assets.delay(node_pk_set, asset_pk_set)
+
+
+@receiver(m2m_changed, sender=ApplicationPermission.system_users.through)
+def on_remote_app_permission_system_users_changed(sender, instance: ApplicationPermission, action, reverse, pk_set, **kwargs):
```
Comment (BaiJiangJie) [inline apps/perms/signals_handler.py:251]: on_application_permission_remote_app_system_users_changed()

File: apps/perms/signals_handler.py [hunk 4]
```diff
@@ -244,3 +245,84 @@ def on_node_asset_change(action, instance, reverse, pk_set, **kwargs):
...
+    if reverse:
+        raise M2MReverseNotAllowed
+
+    if action != POST_ADD:
+        return
+
+    system_users = SystemUser.objects.filter(pk__in=pk_set)
+    logger.debug("Application permission system_users change signal received")
+    attrs = instance.applications.all().values_list('attrs', flat=True)
```
Comment (BaiJiangJie) [inline apps/perms/signals_handler.py:260]: instance.applications.filter(category='remote_app')

File: apps/perms/signals_handler.py [hunk 5]
```diff
@@ -244,3 +245,84 @@ def on_node_asset_change(action, instance, reverse, pk_set, **kwargs):
...
+    assets_id = []
+    for attr in attrs:
+        asset_id = attr.get('asset')
+        if asset_id:
+            assets_id.append(asset_id)
```
Comment (BaiJiangJie) [inline apps/perms/signals_handler.py:265]: assets_id = [attr['asset'] for attr in attrs if attr.get('asset')]

**Pull request merged**